lsp-metals.el

```diff
@@ -574,6 +604,19 @@ WORKSPACE is the workspace we received notification from."
                             'local-map (lsp-metals--status-string-keymap workspace command?))
       workspace)))
 
+(lsp-defun lsp-metals--quick-pick (_workspace (&MetalsQuickPickParams :items :place-holder?))
+  "The Metals quick pick request is sent from the server to the client to let the user provide a string value by picking one out of a number of given options."
+  (let* ((choices (seq-map (lambda (item)
+                             (-let* (((&MetalsQuickPickItem :id :label :description?) item))
+                               (cons (if description? (concat label " " (propertize description? 'face 'font-lock-comment-face)) label) id)))
```

Comment:
I'd suggest to use `affixation-function` from emacs 28 here. Current implementation matches user input by not only item but by description too.

Reply:
I'm sorry, but I don't use Emacs 28 yet. Could you point me in the right direction. I don't know what 
`affixation-function` is and how I could retrofit it to earlier Emacs versions.

Reply:
Here is the commit that adds and explains it: https://github.com/emacs-mirror/emacs/commit/3b740591b0a1d0e7a24be38471499ecace96936b
`affilation-function` is just a property of `completion-extra-properties` that allows to add a prefix/suffix - exactly what we need here. And it should not affect matching against user input. But it will work only in emacs 28, and emacs 27 will just ignore this function.

Reply:
It also should work in emacs 27 with https://github.com/minad/marginalia